### PR TITLE
Make Ethereum transaction handling more resilient

### LINKED ISSUE #7396: Improve resiliency of Ethereum transaction handling
### Problem

The ethereum transaction handling resiliency could be better.

### Solution

Improve it

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by mgoelswirlds**
**Description**:
This PR improves the ethereum transaction handling resiliency.

This PR modifies:
- Exception handling in the EthereumTransactionHandler

**Related issue(s)**:

Fixes #7396

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java [hunk 1]
```diff
@@ -69,29 +71,34 @@ protected void doUpdateTransaction(Transaction transaction, RecordItem recordIte
...
+            ethereumTransaction.setHash(DomainUtils.toBytes(transactionRecord.getEthereumHash()));
+            ethereumTransaction.setMaxGasAllowance(body.getMaxGasAllowance());
+            ethereumTransaction.setPayerAccountId(recordItem.getPayerAccountId());
+
+            entityListener.onEthereumTransaction(ethereumTransaction);
+            updateAccountNonce(recordItem, ethereumTransaction);
+            recordItem.setEthereumTransaction(ethereumTransaction);
+
+            recordItem.addEntityId(ethereumTransaction.getCallDataId());
+        } catch (InvalidDatasetException e) {
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java:98]: The parser can throw any number of exceptions that we should handle.

```suggestion
        } catch (RuntimeException e) {
```

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java [hunk 2]
```diff
@@ -69,29 +71,34 @@ protected void doUpdateTransaction(Transaction transaction, RecordItem recordIte
...
+            Utility.handleRecoverableError(
+                    "Unable to decode ethereum data for transaction at {}", recordItem.getConsensusTimestamp());
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java:100]: This should also print the exception message.

Comment (steven-sheehy) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java:102]: Don't add a new, duplicate log message. Add e.getMessage() to the existing string.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java [hunk 3]
```diff
@@ -69,29 +72,36 @@ protected void doUpdateTransaction(Transaction transaction, RecordItem recordIte
...
+            ethereumTransaction.setPayerAccountId(recordItem.getPayerAccountId());
+
+            entityListener.onEthereumTransaction(ethereumTransaction);
+            updateAccountNonce(recordItem, ethereumTransaction);
+            recordItem.setEthereumTransaction(ethereumTransaction);
+
+            recordItem.addEntityId(ethereumTransaction.getCallDataId());
+        } catch (RuntimeException e) {
+            Utility.handleRecoverableError(
+                    "Error decoding ethereum transaction data at {}",
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java:101]:
```suggestion
                    "Error decoding Ethereum transaction data at {}: {}",
```

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java (hunk 1 above)
Comment (jascks) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java]: It looks like `RuntimeException` works given our exception class hierarchy. Some dependencies could throw `NoSuchElementException` or `IndexOutOfBoundsException`, but they too are `RuntimeException`s.

**Review by xin-hedera — DISMISSED** [review]
LGTM

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java [hunk 4]
```diff
@@ -27,9 +27,12 @@
 import com.hedera.mirror.importer.parser.record.entity.EntityListener;
 import com.hedera.mirror.importer.parser.record.entity.EntityProperties;
 import com.hedera.mirror.importer.parser.record.ethereum.EthereumTransactionParser;
+import com.hedera.mirror.importer.util.Utility;
 import jakarta.inject.Named;
+import lombok.CustomLog;
 import lombok.RequiredArgsConstructor;
 
+@CustomLog
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java:35]: nit: Unused

**Review by steven-sheehy — DISMISSED** [review]

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java [hunk 5]
```diff
@@ -69,29 +72,36 @@ protected void doUpdateTransaction(Transaction transaction, RecordItem recordIte
...
+            entityListener.onEthereumTransaction(ethereumTransaction);
+            updateAccountNonce(recordItem, ethereumTransaction);
+            recordItem.setEthereumTransaction(ethereumTransaction);
+
+            recordItem.addEntityId(ethereumTransaction.getCallDataId());
+        } catch (RuntimeException e) {
+            Utility.handleRecoverableError(
+                    "Error decoding Ethereum transaction data at {}: {}",
+                    recordItem.getConsensusTimestamp(),
+                    e.getMessage());
```
Comment (jascks) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java:103]: I would instead use `e` as the last param to `Utility.handleRecoverableError()`.  This way the logged message will include the exception message and the stack trace. If the halt on error property is set (such as for integration), then the thrown `ParserException` will indicate `e` as the cause and the exception chain is preserved for debugging.

**Review by jascks — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**